### PR TITLE
2078599 add db migrations

### PR DESCRIPTION
**Pull request by JustinHallquist**
This PR adds migrations via `db-migrate` package. There are some issues with initializing the database that are currently being worked on by the author. Until that is finished or until I write a workaround (will make another ticket), circle ci will not be able to use this feature. Will work on a temp fix asap until then.

The scope of this PR is still addressed through these changes regardless of the above issue.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,6 +25,29 @@ Run `ng test` to execute the unit tests via [Karma](https://karma-runner.github.
 Run `ng e2e` to execute the end-to-end tests via [Protractor](http://www.protractortest.org/).
 Before running the tests make sure you are serving the app via `ng serve`.
 
+## Database env setup
+Create two PSQL DBs: loovs_dev, loovs_test
```
Comment (jnhallquist) [inline README.md]: @JustinHallquist Wherever there is `loovs`, please change to `loov'.

Comment (JustinHallquist) [inline README.md]: :+1:

File: example_database.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  "prod": {
+    "driver": "pg",
+    "host": "localhost",
+    "database": "loovs_prod"
```
Comment (jnhallquist) [inline example_database.json]: `loovs_prod` => `loov_prod`

File: README.md [hunk 3]
```diff
@@ -25,6 +25,29 @@ Run `ng test` to execute the unit tests via [Karma](https://karma-runner.github.
...
+copy the example_database.json to database.json (`cp ./example_database.json database.json`)
+replace the usernames in example_database.json with your own psql username
+
+## Creating migrations
+Migrations are handled through the npm package `db-migrate`
+To create a new migration, run
+`db-migrate create migration-name-here --config db/database.json`
+This will add a new file to our migration folder that looks like `./migrations/{timestamp}-migration-name-here.js`
+
+The documentation is awful and you can find all the resources about parameters through people whining on the issues tab of the github project `https://github.com/db-migrate/node-db-migrate`
```
Comment (jnhallquist) [inline README.md]: Probably don't need to trash other people's work/docs.

File: README.md [hunk 4]
```diff
@@ -25,6 +25,29 @@ Run `ng test` to execute the unit tests via [Karma](https://karma-runner.github.
 Run `ng e2e` to execute the end-to-end tests via [Protractor](http://www.protractortest.org/).
 Before running the tests make sure you are serving the app via `ng serve`.
 
+## Database env setup
+Create two PSQL DBs: loov_dev, loov_test
+copy the example_database.json to database.json (`cp ./example_database.json database.json`)
```
Comment (jnhallquist) [inline README.md]: Capitalization

File: README.md [hunk 5]
```diff
@@ -25,6 +25,29 @@ Run `ng test` to execute the unit tests via [Karma](https://karma-runner.github.
...
+replace the usernames in example_database.json with your own psql username
```
Comment (jnhallquist) [inline README.md]: Capitalization

File: README.md [hunk 6]
```diff
@@ -25,6 +25,29 @@ Run `ng test` to execute the unit tests via [Karma](https://karma-runner.github.
...
+To create a new migration, run
+`db-migrate create migration-name-here --config db/database.json`
+This will add a new file to our migration folder that looks like `./migrations/{timestamp}-migration-name-here.js`
+
+The documentation is awful and you can find all the resources about parameters through people whining on the issues tab of the github project `https://github.com/db-migrate/node-db-migrate`
+
+When in doubt, follow the previous migrations as examples
+
+## Running migrations
+make sure database loov_test // loov_dev are created (known bug in db-migration preventing the autocreation for now -- may for and fix)
```
Comment (jnhallquist) [inline README.md]: Capitalization

Comment (jnhallquist) [inline README.md]: `may for and fix` -- huh?

File: README.md [hunk 7]
```diff
@@ -25,6 +25,29 @@ Run `ng test` to execute the unit tests via [Karma](https://karma-runner.github.
...
+
+`db-migrate up -e dev`
+`db-migrate up -e test`
+
+use `db-migrate down --config db/database.json -e test` when you want to revert a migration
```
Comment (jnhallquist) [inline README.md]: Capitalization

File: migrations/20170717035605-create-users-table.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+var dbm;
```
Comment (jnhallquist) [inline migrations/20170717035605-create-users-table.js]: Why `var` instead of `let`?

File: migrations/20170717035605-create-users-table.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+  seed = seedLink;
+};
+
+exports.up = db => {
+  return db.createTable('users', {
+    id: { type: 'int', primaryKey: true, autoIncrement: true },
+    email: { type: 'string', length: 64, unique: true },
+    password: { type: 'string', length: 128 },
+    verified: 'boolean',
+    verification_token: 'string'
```
Comment (jnhallquist) [inline migrations/20170717035605-create-users-table.js]: Should `verification_tokens` be unique?

File: README.md [hunk 10]
```diff
@@ -25,6 +25,29 @@ Run `ng test` to execute the unit tests via [Karma](https://karma-runner.github.
...
+Migrations are handled through the npm package `db-migrate`
+To create a new migration, run
+`db-migrate create migration-name-here --config db/database.json`
+This will add a new file to our migration folder that looks like `./migrations/{timestamp}-migration-name-here.js`
+
+More documentation on the npm package used for migrations at: `https://github.com/db-migrate/node-db-migrate`
+
+When in doubt, follow the previous migrations as examples
+
+## Running migrations
```
Comment (jnhallquist) [inline README.md]: Are both the Creating and Running Migrations sections necessary to include in the readme?

On a related note, what makes information appropriate for the readme vs the repo's wiki?

File: migrations/20170717035605-create-users-table.js (hunk 9 above)
Comment (JustinHallquist) [inline migrations/20170717035605-create-users-table.js]: yes

File: README.md (hunk 10 above)
Comment (JustinHallquist) [inline README.md]: wiki if you want to split things out, say for when the repo starts getting large there may be a 'getting started' page

As for having both, they both do completely different things. It's part of the core functionality for, on one hand setting up the database properly, and on the other hand, adding more tables/indexes/etc

your call.

**Pull request merged**